### ISSUE TITLE
Inquiry: Using ONNX-TensorRT Instead of Native TensorRT API

### ISSUE THREAD

**Issue by alvarofsan** [yuanyang1991/birefnet_tensorrt#3]
Hello,

First of all, thank you for sharing your work—it’s been incredibly helpful!

I was wondering if you’ve considered using the ONNX-TensorRT backend instead of the native TensorRT API for this project. From my understanding,[ ONNX-TensorRT](https://github.com/onnx/onnx-tensorrt) serves as a wrapper for many of the low-level functions that you’ve implemented here, potentially simplifying the code and reducing complexity.

If you explored this option and chose not to go with it, I’d be interested to know more about your reasons or any specific limitations you encountered. Alternatively, if this is something you haven’t yet considered, I’d love to hear your thoughts on the potential advantages or trade-offs.

Thanks once again for all your hard work! I look forward to any insights you can share.

Best regards,
Álvaro.

**Comment by yuanyang1991** [comment]
I previously tried using onnx-tensorRT, but it didn't work in actual tests, possibly due to some specific code within onnx-tensorRT. This is why I didn't choose to use onnx-tensorRT.

**Comment by yuanyang1991** [comment]
If you use some TensorRT-related libraries, like https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy, working with TensorRT becomes quite simple.

**Comment by alvarofsan** [comment]
> If you use some TensorRT-related libraries, like https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy, working with TensorRT becomes quite simple.

Thank you for your response! I’ve been adapting your code to use `pycuda` instead of `cuda`, following the inference class example in this [article](https://tengteng.medium.com/example-inference-code-to-run-tensorrt-10-0-32ea93fdcc2e). I find this approach a bit clearer.

I do have another question regarding the ONNX file—have you tried using dynamic shapes to allow the model to perform batch inference? I’m curious about your experience with that.

**Comment by yuanyang1991** [comment]
Yes, I have used dynamic shapes. If you want to use TensorRT with dynamic shapes as well, you can check out my other GitHub project: https://github.com/yuanyang1991/InferX. It doesn’t even require you to write any pycuda code.

**Comment by alvarofsan** [comment]
Thank you very much. Would it be too much trouble to convert BiRefNet to dynamic shape (as I understand it’s currently set to fixed) and upload the modified .onnx? My GPU doesn’t support running the generalist version. Thanks again!

**Comment by alvarofsan** [comment]
I’ve been exploring your other repository, but I still believe that if the ONNX model is imported with a fixed size, TensorRT won’t be able to build a truly “dynamic engine.” Therefore, it seems necessary to re-export the ONNX model with dynamic_axes. However, I’ve encountered some challenges while trying to do this.
